packages/light-apps/src/Onboarding/Onboarding.tsx

```diff
@@ -8,127 +8,200 @@ import { RouteComponentProps } from 'react-router-dom';
 import keyring from '@polkadot/ui-keyring';
 import { u8aToString } from '@polkadot/util';
 import { mnemonicGenerate, mnemonicToSeed, naclKeypairFromSeed } from '@polkadot/util-crypto';
-import { AddressSummary, Container, FadedText, Input, InputFile, Modal, NavButton, NavLink, Segment, Stacked } from '@polkadot/ui-components';
+import { Container, ErrorText, Modal } from '@polkadot/ui-components';
+import FileSaver from 'file-saver';
 
 import { OnboardingStore } from '../stores/onboardingStore';
+import { AccountStore } from '../stores/accountStore';
+import { CreateNewAccountScreen, ImportOptionsScreen, SaveScreen } from './index';
 
 interface Props extends RouteComponentProps {
   onboardingStore: OnboardingStore;
+  accountStore: AccountStore;
 }
 
-type OnboardingScreenType = 'importOptions' | 'save' | 'new';
+export type OnboardingScreenType = 'importOptions' | 'save' | 'new';
 
 type State = {
   address?: string,
-  isBipBusy: boolean,
+  error: string | null,
+  jsonString?: string,
+  mnemonic: string,
   name?: string,
+  recoveryPhrase?: string,
   password?: string,
-  phrase?: string,
-  screen: OnboardingScreenType,
-  seed?: string
+  screen: OnboardingScreenType
```

Comment:
In general, if we have a state field to manage screens, then we can use React Router (which is just a thin wrapper around this). I know that Onboarding right now is a modal and does not have a route, maybe we could add one?